### PR TITLE
update incremental docs

### PR DESCRIPTION
**Pull request by cgardens**
## What
* User Story: I read that Airbyte now supports "Incremental - Append"; what's that? (this person may not be terribly technical and / or is not looking to spend a lot of time reading dense material.)

## How
* Restructured and reworded incremental docs. My goal is that anyone who hears that we now support incremental will be able to read _from the beginning all of the examples of what incremental looks like_ without it requiring too much thought or technical know how.
* The last two sections source-defined cursor and user defined cursor remain more technical and I am not really expecting anyone to read those sections on their first skim through. I think they still need to be documented for when some sits down to try to understand our data model.

### CODE REVIEW

File: docs/architecture/incremental.md [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+Incremental - Append syncs in Airbyte allow sources to replicate only _new_ or _modified_ data. This prevents re-fetching data that you have already replicated from a source. Going forward, we will call this set of new or updated records the delta.
 
-In this flavor of incremental records in the warehouse will never be deleted or mutated. A new copy of any new or updated records is appended to the data in the warehouse. This means you can find multiple copies of the same record twice in the warehouse and will need to de-duplicate them yourself. We provided an "at least once" guarantee of replicating each record that is present when the sync runs.
+In this flavor of incremental, records in the warehouse will never be deleted or mutated. A copy of each new or updated records is appended to the data in the warehouse. This means you can find multiple copies of the same record twice in the warehouse and will need to de-duplicate them yourself. We provided an "at least once" guarantee of replicating each record that is present when the sync runs.
```
Comment (ChristopheDuong) [inline docs/architecture/incremental.md:7]: in DBT, we could deduplicate rows in the future

Or point to the user how to do it in the DBT models we generate...

**Review by ChristopheDuong — APPROVED** [review]

File: docs/architecture/incremental.md [hunk 2]
```diff
@@ -72,6 +66,16 @@ The output we expect to see in the warehouse is as follows.
...
+
+## User-Defined Cursor
+
+Some sources cannot define the cursor without user input. For example, in the postgres source, the user needs to choose which column in a database table they want to user as the `cursor field`. The author of the source cannot predict this. In these cases the user sets the `cursor_field` in the `ConfiguredAirbyteStream`. (You can find a more detailed description of the configuration data model [here](catalog.md)).
+
+In some cases, the source may propose a `default_cursor_field` in the `AirbyteStream`. When it does, if the user does not specify a `cursor_field` in the `ConfiguredAirbyteStream`, Airbyte will fallback on the default provided by the source. The user is allowed to override the source's `default_cursor_field` by setting the `cursor_field` value in the `ConfiguredAirbyteStream`, but they CANNOT override the `cursor_field` specified in an `AirbyteStream`
```
Comment (michel-tricot) [inline docs/architecture/incremental.md:77]: What info do I get with the last part of the sentence? Is it smth controversial?

**Review by michel-tricot — APPROVED** [review]

File: docs/architecture/incremental.md (hunk 1 above)
Comment (michel-tricot) [reply docs/architecture/incremental.md]: Yeah we should write this tutorial! @cgardens do we already have an issue for that?

File: docs/architecture/incremental.md (hunk 2 above)
Comment (cgardens) [reply docs/architecture/incremental.md]: i'm removing. not relevant to someone just using the UI.

File: docs/architecture/incremental.md (hunk 1 above)
Comment (cgardens) [reply docs/architecture/incremental.md]: no we don't have one yet. please go ahead and make it!

**Pull request merged**